src/crc32c/__init__.pyi

```diff
@@ -1,7 +1,28 @@
 from collections.abc import Buffer
+from typing import Self
 
 big_endian: int
 hardware_based: bool
 
 def crc32(data: Buffer, value: int = 0, gil_release_mode: int = -1) -> int: ...
 def crc32c(data: Buffer, value: int = 0, gil_release_mode: int = -1) -> int: ...
+
+
+class Crc32cHash:
```

Comment:
>but then I realised that collections.abc.Buffer wasn't introduced until 3.12

Yes, this exactly was the reason why I went for a separate stub file because that one is only read by `mypy` (and IDEs).

But, to clean all of this mess up, I implemented some big changes that I will comment on separately.